src/interpreter/mod.rs

```diff
@@ -132,6 +147,64 @@ impl<'txin> Interpreter<'txin> {
         use std::str::FromStr;
         Descriptor::from_str(&self.inferred_descriptor_string())
     }
+
+    /// Returns a sighash over the entire transaction which can be used to verify signatures
+    /// in the descriptor
+    ///
+    /// Not all fields are used by legacy descriptors; if you are sure this is a legacy
+    /// spend (you can check with the `is_legacy` method) you can provide dummy data for
+    /// the amount.
+    pub fn sighash_message(
```

Comment:
Same here. It is not sounding right to call this from  interpreter. Maybe all these methods do belong together but belong in the `Inner`struct. Maybe, we can think of a better name for `Inner`. Maybe `TxComponents`

Reply:
The `Inner` struct is not part of the public API